File: src/test/scala/math/cat/PairTest.java
package math.cat;

import static math.cat.Pair.*;

import junit.framework.TestCase;

/**
 * Tests for math.cat.Pair class
 */
public class PairTest extends TestCase {
  public void testEquals_positive() {
    String s = new StringBuilder("a").append(1 == 2 ? "cb" : "bc").toString();
    assertEquals(Pair("abc", "def"), Pair(s, "def"));
  }

  public void testEquals_negativeX() {
    String s = new StringBuilder("a").append(1 == 1 ? "cb" : "bc").toString();
    assertFalse(Pair("abc", "def").equals(Pair(s, "def")));
  }

  public void testEquals_negativeY() {
    String s = new StringBuilder("a").append(1 == 2 ? "cb" : "bc").toString();
    assertFalse(Pair("abc", "def").equals(Pair(s, "duh...")));
  }
}

File: src/test/scala/math/cat/PairTest.scala
package math.cat

import org.specs2.mutable._
import Pair._

/**
 * Tests for math.cat.Pair class
 */
class PairTest extends Specification {
  "Pair" >> {
    
    "withLeft" >> {
      def sut[Y] = Pair.withLeft[String, Y]("hello")
      sut(42) should_== ("hello", 42)
      sut("world") should_== ("hello", "world")
    }

    "withRight" >> {
      def sut[X] = Pair.withRight[X, String]("hello")
      sut(42) should_== (42, "hello")
      sut("Kitty") should_== ("Kitty", "hello")
    }
    
  }
  
}
